debian/buster/foreman/foreman-plugin.postinst-lib

```diff
@@ -0,0 +1,50 @@
+foreman_plugin_install() {
+  PLUGIN="${1}"
+  PREVIOUS_VERSION="${2}"
+  PLUGIN_HAS_APIPIE="${3:-true}"
+  PLUGIN_HAS_WEBPACK="${4:-true}"
```

Comment:
Can we be smart? In RPMs we have:
https://github.com/theforeman/foreman-packaging/blob/461c8df52c17de853a8a0c0657a7d18135a73229/gem2rpm/foreman_plugin.spec.erb#L20-L24

Should we look for files in `$PLUGIN_ROOT` and determine the values dynamically if the parameter wasn't passed?

Reply:
I guess we don't need the parameter at all in that case ;)

Reply:
So, something like this:

```shell
  # The output sometimes contain additional data like the following line:
  #    Found no changes, using resolution from the lockfile
  # But we really just want the path, so lets grep
  PLUGIN_ROOT="$(${BUNDLE} show ${PLUGIN} | grep '/usr/share/foreman')"
  # Make sure we got a path
  if [ -n "${PLUGIN_ROOT}" ]; then
    if [ -d "${PLUGIN_ROOT}/webpack" ]; then
      mkdir -p $PLUGIN_ROOT/public
      ln -snf /var/lib/foreman/public/webpack/ $PLUGIN_ROOT/public/webpack
    fi
  fi
```

Reply:
Even tho, looking at the actual plugins, those are all plugins that *also* have a `package.json`

Reply:
And calling `apipie:cache:index` doesn't technically hurt for plugins without API, just takes a bit of time.

Reply:
And because I am already talking to myself: I don't think creating this symlink for plugins that don't have package.json/webpack will actually hurt.

Reply:
> I guess we don't need the parameter at all in that case ;)

I thought it could be useful to override. It's better to have that ability now than to need it at some point and having to go through the dance with updating Foreman and then the plugin.

> Even tho, looking at the actual plugins, those are all plugins that _also_ have a `package.json`

Yes. The only odd one used to be somewhere in Katello where there was a package.json but didn't use webpack. For Debian I think we can ignore that edge case.

Reply:
Do you absolutely want the "smart" part?

The current layout works and I don't have the time to play around with the smartness for RC1. :(

Reply:
I'm OK with postponing smartness.

Reply:
I have added it to https://projects.theforeman.org/issues/7988, as it's still "in progress"